public/components/trace_analytics/components/traces/traces_content.tsx

```diff
@@ -273,12 +277,57 @@ export function TracesContent(props: TracesProps) {
       newPageSize,
       setTotalHits,
       props.dataSourceMDSId[0]?.id,
-      sortParams,
+      sort,
       tracesTableMode,
       isUnderOneHour
     ).finally(() => setIsTraceTableLoading(false));
   };
 
+  const refreshTracesTableData = async (
+    sortParams?: { field: string; direction: 'desc' | 'asc' },
+    newPageIndex: number = pageIndex,
+    newPageSize: number = pageSize
+  ) => {
+    setPageIndex(newPageIndex);
+    setPageSize(newPageSize);
+    setIsTraceTableLoading(true);
+    const sort = sortParams ?? getDefaultSort();
+
+    const DSL = filtersToDsl(
+      mode,
+      filters,
+      query,
+      processTimeStamp(startTime, mode),
+      processTimeStamp(endTime, mode),
+      page,
+      appConfigs
+    );
+
+    const timeFilterDSL = filtersToDsl(
+      mode,
+      [],
+      '',
+      processTimeStamp(startTime, mode),
+      processTimeStamp(endTime, mode),
+      page
+    );
+
+    const isUnderOneHour = datemath.parse(endTime)?.diff(datemath.parse(startTime), 'hours')! < 1;
```

Comment:
util functions like this can be extracted, i see the same logic is also used in other places

Reply:
added helper function isUnderOneHourRange
```
export const isUnderOneHourRange = (startTime: string, endTime: string): boolean => {
  const start = dateMath.parse(startTime);
  const end = dateMath.parse(endTime);

  if (!start || !end) return false;

  return end.diff(start, 'hours')! < 1;
};
```